### PR TITLE
Feat: Update let's begin styling (kids-1268)

### PR DESCRIPTION
**Pull request by Daniela510**
## Summary by CodeRabbit

- **New Features**
	- Introduced a customizable `CountryDropDown` widget for enhanced country selection.
	- Updated the `EmailSignupPage` and `FamilyLoginPage` with modern UI components for improved user experience.
	- Added conditional navigation based on user type (U.S. user distinction) in authentication flow.

- **Bug Fixes**
	- Streamlined error handling and dialog presentation for authentication states.

- **Documentation**
	- Added export for new input border styles in shared widgets.

- **Chores**
	- Updated project dependencies by adding `country_flags` for better country representation.

### CODE REVIEW

File: lib/features/auth/widgets/country_dropdown.dart [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import 'package:country_flags/country_flags.dart';
+import 'package:flutter/material.dart';
+import 'package:givt_app/core/enums/country.dart';
+import 'package:givt_app/features/family/utils/utils.dart';
+import 'package:givt_app/l10n/l10n.dart';
+import 'package:givt_app/utils/app_theme.dart';
+
+class CountryDropDown extends StatelessWidget {
```
Comment (TammiLion) [inline lib/features/auth/widgets/country_dropdown.dart:8]: Noice

File: lib/features/children/overview/cubit/family_overview_state.dart [hunk 2]
```diff
@@ -48,12 +48,12 @@ class FamilyOverviewUpdatedState extends FamilyOverviewState {
   List<Profile> sortedAdults(String givtAccountID) {
     return adults
       ..sort((a, b) {
-        final compareNames = a.firstName.compareTo(b.id);
+        final comapreId = a.id.compareTo(b.id);
```
Comment (TammiLion) [inline lib/features/children/overview/cubit/family_overview_state.dart:51]: Just to be sure, why the comparing of IDs? As far as I understand the old code did:
Put whoever is the logged in user first, else compare the first names (so kinda sort alphabetically). The new code does: put whoever is the logged in user first, else order based on id (which to the user is an invisible number).

File: lib/shared/widgets/outlined_input_borders.dart [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import 'package:flutter/material.dart';
+import 'package:givt_app/utils/app_theme.dart';
```
Comment (TammiLion) [inline lib/shared/widgets/outlined_input_borders.dart:2]: Nice!

**Review by TammiLion — APPROVED** [review]
Awesome stuff!

File: lib/features/children/overview/cubit/family_overview_state.dart (hunk 2 above)
Comment (Daniela510) [reply lib/features/children/overview/cubit/family_overview_state.dart:51]: sorry that shouldnt in there

**Pull request merged**